stan/math/prim/scal/prob/neg_binomial_2_log.hpp

```diff
@@ -107,6 +109,11 @@ namespace stan {
         if (include_summand<propto, T_precision>::value)
           logp += lgamma(n_plus_phi[i]);
 
+        // if phi is large we probably overflow, defer to Poisson:
+        if (phi__[i] > 1e12) {
```

Comment:
Is this because as phi gets big, the negative binomial is Poisson like?  If that's right, can you mention that it approaches a Poisson and replace "defer" with "delegate".